scripts/irods/test/test_irsync.py

```diff
@@ -772,3 +772,35 @@ def test_irsync_with_all_keyword_does_not_fail_on_overwrite_issue__issue_8295(se
             f.write('potato')
 
         self.user0.assert_icommand(['irsync', '-a', test_file_path, f'i:{test_data_object_path}'])
+
+    def test_irsync_with_all_keyword(self):
+        data_object_name = 'all_keyword_dataobj'
+        another_resc  = 'resc_all_keyword'
+        another_resc_dir = os.path.join(self.admin.local_session_dir, 'vault_all_keyword')
+        data_object_path = f'{self.admin.session_collection}/{data_object_name}'
+
+        test_file_path = os.path.join(self.testing_tmp_dir, data_object_name)
+
+        with open(test_file_path, 'w') as f:
+            f.write('potato')
+
+        self.admin.assert_icommand(['itouch', data_object_path])
+        self.admin.assert_icommand(['iadmin', 'mkresc', another_resc, 'unixfilesystem', f'{lib.get_hostname()}:{another_resc_dir}'], 'STDOUT_SINGLELINE', 'unixfilesystem')
+
+        try:
+            self.admin.assert_icommand(['irepl', '-R', another_resc, data_object_path])
+            self.admin.assert_icommand(['iput', '-f', test_file_path, data_object_path])
+
+            # Replica should be stale after the iput
+            self.assertTrue(lib.get_replica_status(self.admin, data_object_name, 1) == '0')
+
+            # Edit the file, or irsync will not do the copy
+            with open(test_file_path, 'w') as f:
+                f.write('mashed potato')
+
+            self.admin.assert_icommand(['irsync', '-a', test_file_path, f'i:{data_object_path}'])
+
+            # Replica should be good after the irsync
+            self.assertTrue(lib.get_replica_status(self.admin, data_object_name, 1) == '1')
+        finally:
+            self.admin.run_icommand(['iadmin', 'rmresc', another_resc])
```

Comment:
I imagine you would need to remove the data object first (or at least the replica on this resource) before the resource can be removed.